astra/src/main/java/com/slack/astra/bulkIngestApi/opensearch/BulkApiRequestParser.java

```diff
@@ -72,13 +71,25 @@ public static Trace.Span fromIngestDocument(
 
     // See https://blog.mikemccandless.com/2014/05/choosing-fast-unique-identifier-uuid.html on how
     // to improve this
-    String id =
-        Optional.ofNullable(sourceAndMetadata.get(IngestDocument.Metadata.ID.getFieldName()))
-            .map(String::valueOf)
-            .orElse(UUID.randomUUID().toString());
+    String id = UUID.randomUUID().toString();
```

Comment:
calling `UUID.randomUUID()` could be expensive? Perhaps we should only do it if id is not present?

Reply:
I think this ends up being the default case, but I've went ahead and moved it to only set if not provided.